Library/Homebrew/env_config.rb

```diff
@@ -165,8 +165,20 @@ module EnvConfig
                      "proxying GitHub Packages. If `HOMEBREW_DOCKER_REGISTRY_TOKEN` is set, it will be used instead.",
       },
       HOMEBREW_DOCKER_REGISTRY_TOKEN:            {
-        description: "Use this bearer token for authenticating with a Docker registry proxying GitHub Packages. " \
-                     "Preferred over `HOMEBREW_DOCKER_REGISTRY_BASIC_AUTH_TOKEN`.",
+        description:  <<~EOF,
+          Use this bearer token for authenticating with a Docker registry proxying GitHub Packages.
+          Preferred over `HOMEBREW_DOCKER_REGISTRY_BASIC_AUTH_TOKEN`.
+
+              *Note:* when authenticating against ghcr.io, \
+              the `HOMEBREW_DOCKER_REGISTRY_TOKEN` must be a base64 \
+              encoded GitHub Personal Access Token (PAT).
```

Comment:
Not convinced this description makes that much sense after this? If you're "authenticating against ghcr.io" is this "a Docker registry proxying GitHub Packages"? It doesn't sound like it. Maybe the whole description needs a bigger overhaul?

Reply:
Yea, this is related to the underlying issue I had, trying to access a private tap hosted on a private GitHub repo. And while optimally, there should be a way to be able to assign different tokens for different taps, at the moment this environment variable is the only way to address brew to authenticate again ghcr.io.

Reply:
I wonder if this should default to `HOMEBREW_GITHUB_API_TOKEN` and/or read from the repository URL like https://github.com/Homebrew/brew/pull/16649. Avoiding the user from having to manually run `base64` seems ideal.

Reply:
> should default to `HOMEBREW_GITHUB_API_TOKEN`

While it's highly possibly it's the same token, conceptually it's wrong as the docker registry doesn't have anything to do with the API and vice versa. Wouldn't defaulting to `HOMEBREW_GITHUB_PACKAGES_TOKEN` be more logical (which in itself then might need to be promoted out from development towards more general use)?

> and/or read from the repository URL

I assume it makes sense to allow it to read from the URL, even though having the token in the URL isn't something I personally would ever do.

Reply:
> Wouldn't defaulting to `HOMEBREW_GITHUB_PACKAGES_TOKEN` be more logical (which in itself then might need to be promoted out from development towards more general use)?

Yup, this would make sense!

Reply:
@azatoth Sorry for the confusion here, for more concrete ask here either:
- default to `HOMEBREW_GITHUB_API_TOKEN` and do the Base64 internally inside of Homebrew
or:
- update the description better here as `ghcr.io` is not "a Docker registry proxying GitHub Packages" but instead GitHub Packages itself.

Reply:
> * default to `HOMEBREW_GITHUB_API_TOKEN`

Don't you mean `HOMEBREW_GITHUB_PACKAGES_TOKEN`?

Reply:
No, that's another option we could use but if `HOMEBREW_GITHUB_API_TOKEN` is setup correctly we should be able to Base64 it and use it accordingly.

Reply:
> No, that's another option we could use but if `HOMEBREW_GITHUB_API_TOKEN` is setup correctly we should be able to Base64 it and use it accordingly.

I think I'm a bit confused, I thought the API token was for accessing the API and PACKAGES token to access packages in the event you would want to have different tokes with limited permissions in each

Reply:
`HOMEBREW_GITHUB_PACKAGES_TOKEN` is currently only used for uploading packages.

Reply:
> * default to `HOMEBREW_GITHUB_API_TOKEN` and do the Base64 internally inside of Homebrew

I decided to try this route;
I did rebase against master, but I don't know what your policy for PRs actually is in this regard.